tqdm/_tqdm.py

```diff
@@ -315,6 +392,11 @@ def _decr_instances(cls, instance):
             for inst in cls._instances:
                 if inst.pos > instance.pos:
                     inst.pos -= 1
+            # Kill monitor if no instances are left
+            if not cls._instances and cls.monitor:
+                cls.monitor.exit()
+                del cls.monitor
```

Comment:
No because ```__del__``` cannot be called directly, it's only called by GC. We need a way to signal to the thread that it needs to exit the while loop in `run()`.